### PR TITLE
Add basic frontend

### PR DESCRIPTION
**Pull request by JakubC-projects**
Adds web component with following capabilities:
1. List groups
2. Create groups
3. Edit groups
4. Delete groups

### CODE REVIEW

File: .github/workflows/webcomponent-publish-prod.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - run: |
+          cd ui
+          npm ci --force
+          npm run build
+
+      - id: "auth"
+        name: "Authenticate to Google Cloud"
+        uses: "google-github-actions/auth@v1"
+        with:
+          credentials_json: ${{ secrets.GOOGLE_CREDENTIALS }}
```
Comment (rvanoord) [inline .github/workflows/webcomponent-publish-prod.yml:31]: Should use federated credentials provided by env0 here.
See the esign project for example

File: .github/workflows/webcomponent-publish-sandbox.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      - run: |
+          cd ui
+          npm ci --force
+          npm run build
+
+      - id: "auth"
+        name: "Authenticate to Google Cloud"
+        uses: "google-github-actions/auth@v1"
+        with:
+          credentials_json: ${{ secrets.GOOGLE_CREDENTIALS }}
```
Comment (rvanoord) [inline .github/workflows/webcomponent-publish-sandbox.yml:31]: Use federated credentials

File: frontend/index.html [hunk 3]
```diff
@@ -2,12 +2,14 @@
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
-    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link href="https://fonts.googleapis.com/css2?family=Archivo:wght@200..900&display=swap" rel="stylesheet">
```
Comment (rvanoord) [inline frontend/index.html:6]: Not quite sure this will work in the shaddow dom?

File: frontend/index.html [hunk 4]
```diff
@@ -2,12 +2,14 @@
...
+    <link href="https://fonts.googleapis.com/css2?family=IBM+Plex+Serif&display=swap" rel="stylesheet">
+
     <title>BCC Groups Editor</title>
   </head>
   <body>
-    <div id="app"></div>
+    <group-editor api-base-url="https://en0gzlj9pyza6.x.pipedream.net"/>
```
Comment (rvanoord) [inline frontend/index.html:12]: Shall we prefix all the bcc components with "bcc-"? Little unsure if it was the intention to commit that url?

File: frontend/src/api.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  }
+}
+
+export class StaticTokenSource implements TokenSource {
+  private token;
+  constructor(token: string) {
+    this.token = token;
+  }
+  Token(): Promise<string> {
+    return Promise.resolve(this.token);
```
Comment (rvanoord) [inline frontend/src/api.ts:94]: Will this work with a "reactive" input?

File: frontend/index.html (hunk 4 above)
Comment (JakubC-projects) [reply frontend/index.html:12]: fixed

File: frontend/src/api.ts (hunk 5 above)
Comment (JakubC-projects) [reply frontend/src/api.ts:94]: It didn't, fixed now

File: frontend/index.html (hunk 3 above)
Comment (JakubC-projects) [reply frontend/index.html:6]: The index.html file is only used for viewing it locally. It's not part of the web-component

File: .github/workflows/webcomponent-publish-sandbox.yml (hunk 2 above)
Comment (JakubC-projects) [reply .github/workflows/webcomponent-publish-sandbox.yml]: fixed

File: .github/workflows/webcomponent-publish-prod.yml (hunk 1 above)
Comment (JakubC-projects) [reply .github/workflows/webcomponent-publish-prod.yml]: fixed

**Review by rvanoord — APPROVED** [review]

**Pull request merged**